packages/machines/dialog/src/dialog.machine.ts

```diff
@@ -26,15 +26,24 @@ export function machine(ctx: UserDefinedContext = {}) {
         closeOnOutsideClick: true,
         closeOnEsc: true,
         restoreFocus: true,
+        defaultIsOpen: false,
         ...ctx,
       },
       states: {
         unknown: {
           on: {
-            SETUP: {
-              target: "closed",
-              actions: "setupDocument",
-            },
+            SETUP: [
+              {
+                target: "closed",
+                guard: (ctx) => !ctx.defaultIsOpen,
+                actions: "setupDocument",
+              },
+              {
+                target: "open",
+                guard: (ctx) => !!ctx.defaultIsOpen,
+                actions: "setupDocument",
+              },
+            ],
```

Comment:
I think we can simplify this

```suggestion
            SETUP:  {
                target: ctx.open ? "open": "closed",
                actions: "setupDocument",
              },
```